aea/multiplexer.py

```diff
@@ -567,7 +567,11 @@ async def _send(self, envelope: Envelope) -> None:
         else:
             connection = self._id_to_connection[connection_id]
 
-        connection = cast(Connection, connection)
+        if connection is None:
```

Comment:
This was forgotten in the previous PR which removed the stub connection (and therefore makes it possible to have no connection specified)